### PR TITLE
Add "briefcaseAlias" argument to `acquireNewBriefcaseId`

### PR DESCRIPTION
**Pull request by kabentley**
This will be used to identify who's holding a lock in the case of lock conflicts.

Also, remove unused "codes" methods from `BackendHubAccess` until we figure out final design for code servers.

### CODE REVIEW

**Review by pmconne — COMMENTED** [review]
Is the expectation that in future, acquireNewBriefcaseId will forward the alias to the hub, where it will be stored and from where it can be queried?

**Comment by kabentley** [comment]
> Is the expectation that in future, acquireNewBriefcaseId will forward the alias to the hub, where it will be stored and from where it can be queried?

The IModelHub implementation of `BackendHubAccess.acquireNewBriefcaseId` will send this parameter to the server where it will be stored along with the briefcaseId's owner (there will be a default value). Whenever their implementation of `BackendHubAccess.acquireLocks` returns a conflict, it will return the alias and throw an exception (see HubMocks' reference implementation.)

I'm waiting for someone from their team to give me their GitHub name so i can add them as a reviewer (I've talked to them about this already).

File: common/api/imodeljs-backend.api.md [hunk 1]
```diff
@@ -2887,6 +2888,15 @@ export class LocalhostIpcHost {
     }): Promise<void>;
 }
 
+// @beta
+export class LockConflict extends IModelError {
+    constructor(
+    briefcaseId: BriefcaseId,
+    briefcaseAlias: string, msg: "shared lock is held" | "exclusive lock is already held");
```
Comment (robertas-kerpys) [inline common/api/imodeljs-backend.api.md:2895]: Should this use `message` property of `BentleyError `class? I do not see `msg` property in the inheritance chain.

**Review by robertas-kerpys — COMMENTED** [review]
I do not yet have scheduled the work to implement briefcase alias support in iModels REST API. It probably won't be available until end of next iteration. From the code I see that it is not required to be immediately available in iModels REST API so it should be ok.

File: core/backend/src/BackendHubAccess.ts [hunk 2]
```diff
@@ -29,6 +28,21 @@ export enum LockState {
...
+*/
+export class LockConflict extends IModelError {
+  public constructor(
+    /** Id of Briefcase holding lock */
+    public readonly briefcaseId: BriefcaseId,
+    /** Alias of Briefcase holding lock */
+    public readonly briefcaseAlias: string,
+    msg: "shared lock is held" | "exclusive lock is already held"
+  ) {
+    super(IModelHubStatus.LockOwnedByAnotherBriefcase, msg);
```
Comment (kabentley) [inline core/backend/src/BackendHubAccess.ts:42]: @robertas-kerpys see here

File: common/api/imodeljs-backend.api.md (hunk 1 above)
Comment (kabentley) [reply common/api/imodeljs-backend.api.md:2895]: it does. It is a bit confusing to look at these .md files. They're generated for the api checker and only show the signatures/properties.

**Comment by kabentley** [comment]
> I do not yet have scheduled the work to implement briefcase alias support in iModels REST API. It probably won't be available until end of next iteration. From the code I see that it is not required to be immediately available in iModels REST API so it should be ok.

That's fine. The editing applications that will use this are being created now, but don't need this to proceed.

**Review by robertas-kerpys — APPROVED** [review]

**Pull request merged**